compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1482,10 +1482,10 @@ impl SourceScope {
         let scope_data = &source_scopes[self];
         if let Some((inlined_instance, _)) = scope_data.inlined {
             Some(inlined_instance)
-        } else if let Some(inlined_scope) = scope_data.inlined_parent_scope {
-            Some(source_scopes[inlined_scope].inlined.unwrap().0)
         } else {
-            None
+            scope_data
+                .inlined_parent_scope
+                .map(|inlined_scope| source_scopes[inlined_scope].inlined.unwrap().0)
```

Comment:
I have mixed feelings about `manual_map` in general, but this particular pattern of replacing only the tail of an if-chain with a `map` seems like a clear downgrade to me. 

Now the final condition (using `map`) is inconsistent with the prior conditions (which are still using `if let Some` followed by `Some`), for no real benefit.

Reply:
+1 I find the replacing only the tail with a `map` makes it quite hard to read